src/main/c/zlib/zlib.c

```diff
@@ -1978,7 +1978,9 @@ static VALUE
 rb_inflate_add_dictionary(VALUE obj, VALUE dictionary)
 {
     VALUE dictionaries = rb_ivar_get(obj, id_dictionaries);
-    VALUE checksum = do_checksum(1, &dictionary, adler32);
+    VALUE argv[1];
```

Comment:
I'm not a big fan of directly patching the source. Can we handle this with `preproces.rb`? If not, I think we should at least comment what we've changed and why.

Reply:
This looks like a fix that `preprocess.rb` already applies.

Patches are ok in the source, because we have a mechanism for making a patch of them and re-applying, but `preprocess.rb` is better.

Reply:
I thought we have a distinction between gems and stdlib: stdlib C files are modified in place and gem C files go into `preprocess.rb`. `zlib.c` in particular has a lot of modifications already (mostly handles), therefore I don't think it makes sense to move just this one, all the modifications would have to be moved. However I think it's better to have the modifications for stdlib files in place it's much more manageable then `preprocess.rb`.

I'll add a short comment.

Reply:
`preprocess.rb` indeed applies similar fix but only specifically for JSON gem.

Reply:
My opinion: the handles would be more succinctly expressed in regexps and maybe catch more cases easily, but there is also the risk to have buggy or incomplete regexps.
Direct patching is likely to incur more conflicts when updating, but hopefully the zlib cext doesn't change much between Ruby versions.

Once `zlib` becomes a proper gem (in 2.5) we will not have a choice if we want to support installing zlib with RubyGems. But that's far ahead.

Reply:
zlib.c has about 10 commits per year usually small changes, it's quite stable.